src/js/Cell.js

```diff
@@ -16,22 +17,41 @@ class Cell extends MazeEntity {
      * @returns {Array.<Cell>} List of neighboring cells
      */
     getNeighbors() {
-
+        return this.getNeighboringEntitiesList(2)
     }
 
     /**
-     * @param {Cell} cell 
+     * @param {Cell} cell
      * @returns {boolean} Represents whether there is a removed wall between cells
      */
     canAccessCell(cell){
+      return (
+          cell.visited==false
+          && this.isNeighboringCell(cell)
+          && this.getInterveningWall(cell).state === WallState.PENDING
+      )
+    }
 
+    getInterveningWall(cell){
+      if(this.isNeighboringCell(cell)){
+        return this.maze.data[(this.row+cell.row)/2][(this.col+cell.col)/2]
+      }
+      return null;
+    }
+
+    isNeighboringCell(cell){
+      const isNeighboringCoordinateSet = (x1,x2,y1,y2) => (x1==x2 && Math.abs(y1-y2) == 2)
+      return (
+          isNeighboringCoordinateSet(this.row, cell.row, this.col, cell.col)
+          || isNeighboringCoordinateSet(this.col, cell.col, this.row, cell.row)
+      )
     }
 
     /**
      * @returns {Array.<Cell>} List of neighboring cells that are accessible
      */
     getAccessibleNeighbors(){
-
+      return this.getNeighbors().filter((c)=>c&&this.canAccessCell(c))
```

Comment:
can you help me understand why we need to check for the existence of `c`? would `getNeighbors` ever return a null entity?

Reply:
(never mind, I just looked at your implementation, and now I understand... I feel like maybe we should filter null entities out of the list, though, before `getNeighboringEntitiesList` returns?)